src/uint/shl.rs

```diff
@@ -94,23 +104,28 @@ impl<const LIMBS: usize> Uint<LIMBS> {
     /// Computes `self >> 1` in constant-time.
     pub(crate) const fn shl1(&self) -> Self {
         // TODO(tarcieri): optimized implementation
```

Comment:
Sidebar, but I've been wondering if we could implement this using a more optimized overflowing shift on the level of each limb, so as to better optimize carry bit handling (are there e.g. CPU flags which make retrieving overflow possible?)

Reply:
I tried that (using `overflowing_shift()`, which is supposed to compile down to a shift + flag checking), and it doesn't really change the performance. Plus I am more confident that the explicit overflow will compile into constant-time code, because if there are architectures where there's no overflow flag, the compiler will probably just put in a condition.

Reply:
Better guarantees of constant-time are a good point, but it might still be good to have something like `Limb::overflowing_sh*` to provide a common implementation in places where this is useful, as it would be here, regardless of how it's implemented.

In the implementation of the function on `Limb`, we could potentially decide on an architecture-by-architecture basis to use the primitive methods if we're happy with the generated assembly, and have the portable implementation fall back to one we hope will be mostly constant time-ish.